### PR TITLE
DP-548 Fix sse retry

### PR DESCRIPTION
**Pull request by Ronserruya**
Fixes two issues

1. DP-548, an issue when sleeping before retrying SSE
2. DP-577, a bug where we retired a tx 25 times instead of 5 times.

### CODE REVIEW

File: kin_base/horizon.py [hunk 1]
```diff
@@ -113,7 +113,6 @@ def __init__(self,
     async def __aexit__(self, exc_type, exc_val, exc_tb) -> None:
         await self.close()
 
-    @_retry
```
Comment (yosriz) [inline kin_base/horizon.py:116]: the bug is that the retry happens in `kin-base` and it's not needed here?

File: kin_base/horizon.py [hunk 2]
```diff
@@ -216,7 +215,7 @@ def __init__(self,
                                 # Events that dont have an id are not useful for us (hello/byebye events)
                                 # Save the last event id and retry time
                                 last_id = event.last_event_id
-                                retry = client._reconnection_time
+                                retry = client._reconnection_time.total_seconds()
```
Comment (yosriz) [inline kin_base/horizon.py:218]: what's `reconnection_time`? and what's `total_seconds` ?

File: kin_base/horizon.py [hunk 3]
```diff
@@ -225,7 +224,7 @@ def __init__(self,
                 except aiohttp.ClientPayloadError:
                     # Retry if the connection dropped after we got the initial response
                     logger.debug('Resetting SSE connection for {} after timeout'.format(url))
-                    await asyncio.sleep(retry.total_seconds())
+                    await asyncio.sleep(retry)
```
Comment (yosriz) [inline kin_base/horizon.py:227]: what's the fix here?

File: kin_base/horizon.py (hunk 1 above)
Comment (Ronserruya) [reply kin_base/horizon.py:116]: This is kin_base

The bug is that _retry happened on ```submit``` and on ```_post```, but since ```submit``` uses ```_post```, the retry happened 5^2 times

File: kin_base/horizon.py (hunk 2 above)
Comment (Ronserruya) [reply kin_base/horizon.py:218]: ```_reconnection_time``` is the time to sleep before renewing the connection, its in milliseconds so ```.total_seconds``` converts it

File: kin_base/horizon.py (hunk 3 above)
Comment (Ronserruya) [reply kin_base/horizon.py:227]: We set ```retry``` to 0.1 at the start, so the bug was that if we never got an answer from the server, ```retry``` stayed a float, and didn't had the ```total_seconds()``` method. So now I do the conversion when I get the retry from the server, and not at the end

File: kin_base/horizon.py (hunk 1 above)
Comment (yosriz) [reply kin_base/horizon.py:116]: yep, silly me, anyway got it

File: kin_base/horizon.py (hunk 2 above)
Comment (yosriz) [reply kin_base/horizon.py:218]: 👍

File: kin_base/horizon.py (hunk 3 above)
Comment (yosriz) [reply kin_base/horizon.py:227]: 👍

**Review by yosriz — APPROVED** [review]
@Ronserruya can you try to add tests for retry? a good opportunity to add unit tests
(sse is probably harder than simple query from my experience)

**Comment by Ronserruya** [comment]
@yosriz Added tests for retry (and fixed some issues with it as well :) )
Pls review again

File: tests/test_horizon.py [hunk 4]
```diff
@@ -85,4 +90,20 @@
...
+    async with Horizon(setup.horizon_endpoint_uri) as horizon:
+        horizon._session.get = MagicMock(side_effect=ClientConnectionError)
+        start = time.time()
+        with pytest.raises(HorizonRequestError):
+            await horizon.account('GA3FLH3EVYHZUHTPQZU63JPX7ECJQL2XZFCMALPCLFYMSYC4JKVLAJWM')
+
+        elapsed = time.time() - start
+        # Sum of arithmetic progression
+        a_n = horizon.num_retries - 1
+        expected_time = ((2 * horizon.backoff_factor + (a_n - 1) * horizon.backoff_factor) *
```
Comment (yosriz) [inline tests/test_horizon.py]: not sure I understand why it's calculated dynamically, in this case, there should be a single retry? if that the case then hardcode the expected time instead of calculating it (cause if for instance `horizon. num_retries` is lying - test will pass)
for completeness, I would add a test for 5 retry failure with the expectation of total failure. (test the edge cases)
